tests/typescript_generic/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -130,6 +130,56 @@ export const getVehicleDescriptor = async (
 ================================================================================
 `;
 
+exports[`empty.ts 1`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+f1<>();
+
+new f2<>();
+
+function f3<>() {}
+
+class f4 {
+    constructor<>() {}
```

Comment:
Constructors can't have type parameters, but classes (and interfaces) can.

Reply:
@thorn0 can :smile: https://github.com/typescript-eslint/typescript-eslint/pull/223/files#diff-a0bd6d0e4d604ca222ff0af725e5b63fR2

Reply:
I think we need a separate folder for tests like this. `typescript_invalid or `typescript_error_recovery`.

Reply:
Okey, let's remove that from the PR

Reply:
You misunderstood me. **All** this code is not valid TS. Not only constructors.